src/Arcus.Messaging.Tests.Integration/MessagePump/ServiceBusMessagePumpTests.cs

```diff
@@ -185,6 +191,48 @@ await using (var service = await TestMessagePumpService.StartNewAsync(config, _l
             }
         }
 
+        [Fact]
+        public async Task ServiceBusMessagePump_FailureDuringMessageHandling_TracksCorrelationInApplicationInsights()
+        {
+            // Arrange
+            string operationId = $"operation-{Guid.NewGuid()}", transactionId = $"transaction-{Guid.NewGuid()}";
+
+            var config = TestConfig.Create();
+            ApplicationInsightsConfig applicationInsightsConfig = config.GetApplicationInsightsConfig();
+            string connectionString = config.GetServiceBusConnectionString(ServiceBusEntity.Queue);
+            var commandArguments = new[]
+            {
+                CommandArgument.CreateSecret("APPLICATIONINSIGHTS_INSTRUMENTATIONKEY", applicationInsightsConfig.InstrumentationKey),
+                CommandArgument.CreateSecret("ARCUS_SERVICEBUS_CONNECTIONSTRING", connectionString)
+            };
+
+            using var project = await ServiceBusWorkerProject.StartNewWithAsync<ServiceBusQueueTrackCorrelationOnExceptionProgram>(config, _logger, commandArguments);
+            await using var service = await TestMessagePumpService.StartNewAsync(config, _logger);
+            Message orderMessage = OrderGenerator.Generate().AsServiceBusMessage(operationId, transactionId);
+                    
+            // Act
+            await service.SendMessageToServiceBusAsync(connectionString, orderMessage);
+
+            // Assert
+            using (ApplicationInsightsDataClient client = CreateApplicationInsightsClient(applicationInsightsConfig.InstrumentationKey))
+            {
+                await RetryAssertUntilTelemetryShouldBeAvailableAsync(async () =>
+                {
+                    const string onlyLastHourFilter = "timestamp gt now() sub duration'PT1H'";
+                    EventsResults<EventsExceptionResult> results = 
+                        await client.Events.GetExceptionEventsAsync(applicationInsightsConfig.ApplicationId, filter: onlyLastHourFilter);
+
+                    Assert.Contains(results.Value, result =>
+                    {
```

Comment:
Can we also assert that the operation & transaction id are assigned to the correct AI fields, such as `operation.id` for operation id?

This requires our sink of course.

Reply:
Yeah, we can do that. We're already using our sink here, so that's OK.

Reply:
Voila ✔️

Reply:
The transaction ID not yet set, as we've discussed.

Reply:
💘